### ISSUE TITLE
BUG: Fix source code flash on load

### ISSUE THREAD

**Issue by chrisjm** [chrisjm/openbrewerydb-gatsby#9]
# Bug Report

## Describe the bug

When loading the page in Firefox and Chrome, you see all of the source code for a split-second before the website loads up.

## Steps to reproduce bug

1. Go to https://www.openbrewerydb.org/
2. Reload the page

### Expected behavior

The site should load up without showing the source code flash.

## Screenshots

<img width="1440" alt="Screenshot 2019-12-30 09 37 52" src="https://user-images.githubusercontent.com/96110/71593241-17e6c380-2ae8-11ea-93d8-a6291753a141.png">

### Desktop

- OS: Mac OS X
- Browser: Chrome and Firefox (probably others)

## Additional context

This started happening after running through #3. There might be something related to the changes made there (and there were a lot of changes) which affected the rendering. The starter theme I used, [gatsby-gitbook-starter](https://www.gatsbyjs.org/starters/hasura/gatsby-gitbook-starter/) is a mess and probably should be built from the ground-up.

**Comment by teisenhower** [comment]
Hey @chrisjm! 

I started looking into this and I can definitely replicate the issue but only in a production environment after a `gatsby build`, it does not happen while in`gatsby develop`

I found someone in the Gatsby repo experiencing the same issue, [#19061](https://github.com/gatsbyjs/gatsby/issues/19061). Unfortunately the thread doesn't provide a solid solution, the last comment hints at it being a possible issue with the hosting..?

Things I was able to find:

- The page is only displaying the code that lives inside the root `id="___gatsby"` div.

- In dev tools I set the speed to Slow 3G so I could watch as assets were loaded, it looks like everything finally renders properly when [openbrewerydb-gatsby](https://api.github.com/repos/chrisjm/openbrewerydb-gatsby) gets loaded. Not sure why that would be but that behavior is consistent.  

![openDB](https://user-images.githubusercontent.com/47988669/74558992-54a03a00-4f31-11ea-851f-49980bbba461.gif)

I will continue debugging, just wanted to throw the information I was able to uncover your way in hopes it would help in some way!

Happy Friday! :beers:

**Comment by chrisjm** [comment]
@teisenhower Great findings! Definitely weird re: rendering after openbrewerydb-gatsby loads. 🤔 

Thanks for your work and time on this. Let me know how I can help.

**Comment by teisenhower** [comment]
Hey @chrisjm!

Hope this message finds you well. I have a little more information for you that I hope will help. 

After looking through the commit history it looks like commit b9206ab4cf3b238ad08b39079043f83957216750 is where this started happening. All commits prior to this do not have the flash.

And I believe I may have found the main cause. In `html.js`, the body is being loaded as a string 
https://github.com/chrisjm/openbrewerydb-gatsby/blob/1527e7415dbe213ef5086a94d46c70a155ada5ef/src/html.js#L59-L61

https://github.com/chrisjm/openbrewerydb-gatsby/blob/1527e7415dbe213ef5086a94d46c70a155ada5ef/src/html.js#L73

This is what you’re seeing in the browser. All your HTML displayed just as a straight string. It is not until after your Javascript is loaded that the page is then fully rendered. I was able to determine this by disabling Javascript which resulted in the page being stuck on just the string of HTML

##### Disabling JS
![disableJS](https://user-images.githubusercontent.com/47988669/78508827-d6555c80-7757-11ea-9dff-796c12084b92.gif)


I was able to fix this flash of text by reverting the `main id="___gatsby"` div back to using `dangerouslySetInnerHTML`

```jsx
<div
   key="body"
   id="___gatsby" 
   dangerouslySetInnerHTML={{ __html: body }}>
</div>
```

![fouc](https://user-images.githubusercontent.com/47988669/78510111-ec1b4f80-7760-11ea-973c-33aac73e7a7d.gif)

This unfortunately comes with its own slight issue. A Flicker Of Unstyled Content (*or 'FOUC' as you'll find it on forums and Gatsby issues because we know developers love acronyms*). From what I understand, this has to do with server side rendering and how that works when using CSS-in-JS libraries such as Emotion. The content is partially unstyled until the Javascript has fully loaded.

##### The [Gatsby Docs](https://www.gatsbyjs.org/docs/css-in-js/) hint at this 'fouc' 
> Keep in mind that styles aren’t applied until the JavaScript loads hence a plugin to extract the styles is necessary to prevent flash of unstyled content (FOUC). To cater for this, every CSS-in-JS library has a Gatsby plugin which you need to extract styles and insert them into the HTML during builds and this prevents FOUC.

I didn't want to open a PR for the flash issue until you had a chance to look this over since it unfortunately brings another bug to the surface. I started looking into this as well and have already been able to remedy some of the unstyled content issues but I wanted to get the information about the code flash to you first.

Thanks! 
:metal:

**Comment by chrisjm** [comment]
Great investigation work @teisenhower! I remember changing that `dangerouslySetInnerHTML` because of an ESLint warning. I was updating a lot of code during that time. Whoops. 🤦‍♂ 

Definitely submit a PR to fix the code flash. If you need to turn off ESLint for that file or line, feel free to do so.

We can fix the FOUC in a subsequent issue.

Thank you so much for all of your hard work! 💯

**Comment by teisenhower** [comment]
Thanks @chrisjm! No problem! Happy to help out!

PR has been submitted. I had to do a second commit, I forgot to remove `{body}` from within the div.. :confused: sorry for that..  

I will keep working on the FOUC as well.

Thanks again! :metal:

**Comment by chrisjm** [comment]
Fixed with #17

**Comment by teisenhower** [comment]
Hey @chrisjm!

Fixed the FOUC!

Would you like me to open a new issue and submit a PR?

![open-brew-db](https://user-images.githubusercontent.com/47988669/79054451-e915d880-7c12-11ea-8d91-0e7654fb1d00.gif)

Hope all is well!

**Comment by chrisjm** [comment]
@teisenhower You are awesome! I created a new issue #20 for you. Looking forward to the PR!

**Comment by teisenhower** [comment]
@chrisjm Thanks!

I am getting another linting error `Unable to resolve path to module` and all the solutions I have read involve updating the `.eslintrc.json`. I didn't want to do that without running it by you first.

**Comment by chrisjm** [comment]
@teisenhower Weird. I'm not against updating the linter config. Go ahead and submit a PR and include a link to the solutions you found and I will review. 🍻
